internal/alertmanager/upstream.go

```diff
@@ -78,3 +75,19 @@ func GetAlertmanagerByName(name string) *Alertmanager {
 	}
 	return nil
 }
+
+// WithProxy option can be passed to NewAlertmanager in order to enable request
+// proxying for unsee clients
+func WithProxy(proxied bool) Option {
+	return func(am *Alertmanager) {
+		am.ProxyRequests = proxied
+	}
+}
+
+// WithRequestTimeout option can be passed to NewAlertmanager in order to set
+// a custom timeout for Alertmanager upstream requests
+func WithRequestTimeout(timeout time.Duration) Option {
+	return func(am *Alertmanager) {
+		am.Timeout = timeout
```

Comment:
Any reason not to change the struct member name as well? Any reason to set a request timeout rather than using a `context.WithTimeout`?

Reply:
1. None, renamed
2. I know about context but didn't really use it for anything so lack any experience and obvious patterns I could use it for